bundles/org.openhab.binding.yamahamusiccast/src/main/java/org/openhab/binding/yamahamusiccast/internal/YamahaMusiccastHandler.java

```diff
@@ -0,0 +1,1238 @@
+    private Gson gson = new Gson();
+    private Logger logger = LoggerFactory.getLogger(YamahaMusiccastHandler.class);
+    private @Nullable ScheduledFuture<?> keepUdpEventsAliveTask;
+    private @Nullable ScheduledFuture<?> generalHousekeepingTask;
+    private @Nullable YamahaMusiccastConfiguration config;
+    private @Nullable String httpResponse;
+
+    private @Nullable JsonParser parser = new JsonParser();
+    @Nullable
+    String tmpString = "";
+    int tmpInteger = 0;
+    int volumePercent = 0;
+    int volumeAbsValue = 0;
+    int connectionTimeout = 5000;
+    int longConnectionTimeout = 60000;
+    @Nullable
+    String responseCode = "";
+    @Nullable
+    String powerState = "";
+    @Nullable
+    String muteState = "";
+    int volumeState = 0;
+    int maxVolumeState = 0;
+    @Nullable
+    String inputState = "";
+    int presetNumber = 0;
+    @Nullable
+    String soundProgramState = "";
+    int sleepState = 0;
+    @Nullable
+    String playbackState = "";
+    @Nullable
+    String artistState = "";
+    @Nullable
+    String trackState = "";
+    @Nullable
+    String albumState = "";
+    @Nullable
+    String albumArtUrlState = "";
+    @Nullable
+    String repeatState = "";
+    @Nullable
+    String shuffleState = "";
+    int playTimeState = 0;
+    int totalTimeState = 0;
+    @Nullable
+    String topicAVR = "";
+    @Nullable
+    String zone = "main";
+    String channelWithoutGroup = "";
+    @Nullable
+    String thingLabel = "";
+    @Nullable
+    String mclinkSetupServer = "";
+    @Nullable
+    String mclinkSetupZone = "";
+    String url = "";
+    String json = "";
+    String action = "";
+
+    int zoneNum = 0;
+    @Nullable
+    String groupId = "";
+    @Nullable
+    String role = "";
+    @Nullable
+    String host;
+    @Nullable
+    public String deviceId = "";
+
+    private YamahaMusiccastStateDescriptionProvider stateDescriptionProvider;
+
+    public YamahaMusiccastHandler(Thing thing, YamahaMusiccastStateDescriptionProvider stateDescriptionProvider) {
+        super(thing);
+        this.stateDescriptionProvider = stateDescriptionProvider;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command != RefreshType.REFRESH) {
+            logger.debug("Handling command {} for channel {}", command, channelUID);
+            channelWithoutGroup = channelUID.getIdWithoutGroup();
+            zone = channelUID.getGroupId();
+            DistributionInfo distributioninfo = new DistributionInfo();
+            Response response = new Response();
+            switch (channelWithoutGroup) {
+                case CHANNEL_POWER:
+                    if (command == OnOffType.ON) {
+                        httpResponse = setPower("on", zone);
+                        response = gson.fromJson(httpResponse, Response.class);
+                        tmpString = response.getResponseCode();
```

Comment:
`tmpString` seems rather a local variable than a field.
You need to check the return value of `fromJson()` as it can be null.

Please check all.

Reply:
There was no check in the DTO, this has been added.
Will be available with next push.